x/pairing/client/cli/tx_simulate_relay_payment.go

```diff
@@ -0,0 +1,211 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/utils/sigs"
+	"github.com/lavanet/lava/x/pairing/types"
+	"github.com/spf13/cobra"
+)
+
+var _ = strconv.Itoa(0)
+
+const (
+	QoSValuesFlag = "qos-values"
+	CuAmountFlag  = "cu-amount"
+	EpochFlag     = "epoch"
+	RelayNumFlag  = "relay-num"
+)
+
+func CmdSimulateRelayPayment() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "simulate-relay-payment [consumer-key] [spec-id]",
+		Short: "Create & broadcast message relayPayment for simulation",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			// Extract clientCtx
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			// Extract arguments
+			consumerAddress := args[0]
+
+			keyName, err := sigs.GetKeyName(clientCtx.WithFrom(consumerAddress))
+			if err != nil {
+				return err
+			}
+
+			privKey, err := sigs.GetPrivKey(clientCtx, keyName)
+			if err != nil {
+				return err
+			}
+
+			// SpecID
+			specId := args[1]
+
+			// CU
+			cuAmount, err := cmd.Flags().GetUint64("cu-amount")
+			if err != nil {
+				return err
+			}
+
+			// Provider
+			providerAddr := clientCtx.GetFromAddress().String()
+
+			// Relay Num
+			relayNum, err := cmd.Flags().GetUint64(RelayNumFlag)
+			if err != nil {
+				return err
+			}
+
+			// QoS
+			qosValues, err := cmd.Flags().GetStringSlice(QoSValuesFlag)
+			if err != nil {
+				return err
+			}
+			qosReport, err := extractQoSFlag(qosValues)
+			if err != nil {
+				return err
+			}
+
+			// Epoch
+			epochValue, err := cmd.Flags().GetUint64(EpochFlag)
+			if err != nil {
+				return err
+			}
+			epoch, err := extractEpoch(clientCtx, epochValue)
+			if err != nil {
+				return err
+			}
+
+			relaySessions := []*types.RelaySession{}
+			for relayIdx := uint64(1); relayIdx <= relayNum; relayIdx++ {
+				// Session ID
+				// We need to randomize it, otherwise it will give unique ID error
+				sessionId := uint64(time.Now().UnixNano())
+
+				// Create RelaySession
+				relaySession := newRelaySession(specId, sessionId, cuAmount, providerAddr, relayIdx, qosReport, epoch, clientCtx.ChainID)
+
+				sig, err := sigs.Sign(privKey, *relaySession)
+				if err != nil {
+					return err // consider whether you want to return or continue to the next iteration
+				}
+				// Set Sig
+				relaySession.Sig = sig
+
+				relaySessions = append(relaySessions, relaySession)
+			}
+
+			msg := types.NewMsgRelayPayment(
+				clientCtx.GetFromAddress().String(),
+				relaySessions,
+				"",
```

Comment:
lets add a description saying this is a simulation

Reply:
fixed here: 922c3306a48906735deaec1c07fbb036fcc788ce